library/alloc/tests/arc.rs

```diff
@@ -210,3 +210,18 @@ fn weak_may_dangle() {
     // `val` dropped here while still borrowed
     // borrow might be used here, when `val` is dropped and runs the `Drop` code for type `std::sync::Weak`
 }
+
+#[test]
+fn arc_from_vec_opt() {
```

Comment:
I think there is UB in this test: https://github.com/rust-lang/rust/issues/104565.